python/hsfs/core/statistics_engine.py

```diff
@@ -69,9 +79,51 @@ def compute_statistics(
             # Python engine
             engine.get_instance().profile_by_spark(metadata_instance)
 
+    def compute_single_feature_statistics(
+        self,
+        feature_dataframe,
+        feature_name,
+    ) -> FeatureDescriptiveStatistics:
+        """Compute statistics for a single feature without sending the result to Hopsworks.
+
+        Args:
+            feature_dataframe: Single-feature Spark or Pandas DataFrame to compute the statistics on.
+            feature_name: str. Name of the feature.
+
+        Returns:
+            FeatureDescriptiveStatistics. Descriptive statistics of the given feature dataframe.
+        """
+
+        # TODO: Future work. Persisting the statistics and returning the stats together with the ID
+
+        if engine.get_type() == "spark":
+            statistics_str = StatisticsEngine.profile_statistics(
+                feature_dataframe, [feature_name], False, False, False
+            )
+            return FeatureDescriptiveStatistics.from_deequ_json(statistics_str)
+        else:
+            # TODO: Only compute statistics with Spark at the moment. This method is expected to be called
+            # only through run_feature_monitoring() which is the entrypoint of the feature monitoring job.
+            # Pending work is to use Python jobs if the dataframe is relatively small.
```

Comment:
Not really python does not support incremental queries and you would have to fetch the whole feature view data instead of a specific column so I don't think it is a going to happen to move to python

Reply:
You are right about the incremental queries. Mentioning "Python jobs" was misleading, my bad.
I was thinking about the cases we can compute feature statistics directly in the Python client. For example for training datasets, batch data or FM on every insert. This method will be called from both Spark job or Python client, but for now it only works for Spark jobs.
I will update the comment.